### PR TITLE
[XM] Fix 44707 by fixing launcher and mmp

### PR DESCRIPTION
**Pull request by chamons**
Also includes some test infrastructure improvements that can split out if necessary.

### CODE REVIEW

**Comment by chamons** [comment]
@kumpera - This is the XM side of 44707. We're not generating an "empty" machine.config since you fixed mono, but the option is there --custom-machine-config="" in case the bug resurfaces.

@timrisi Please look over the test changes.

@rolfbjarne The launcher changes are trivial, but can you verify they are in the right place?

**Comment by chamons** [comment]
The XM_TEST_NAME hack is something i've wanted to do for awhile, so I can stop locally hacking that file every time one test fails.

File: runtime/launcher.m [hunk 1]
```diff
@@ -357,6 +358,8 @@
 		// So route through MONO_DEBUG
 		setenv ("MONO_DEBUG", "no-gdb-backtrace", 0);
 	}
+
+	setenv ("MONO_CFG_DIR", [monobundle_dir UTF8String], 1);
```
Comment (rolfbjarne) [inline runtime/launcher.m]: We already call `mono_set_dirs` [1], whose second parameter sets the config directory. Maybe we should check why that doesn't work instead?

[1] https://github.com/xamarin/xamarin-macios/blob/master/runtime/launcher.m#L495-L499

File: tools/mmp/driver.cs [hunk 2]
```diff
@@ -287,7 +288,8 @@ static void Main2 (string [] args)
 				{ "http-message-handler=", "Specify the default HTTP Message Handler", v => { http_message_provider = v; }},
 				{ "extension", "Specifies an app extension", v => is_extension = true },
 				{ "allow-unsafe-gac-resolution", "Allow MSBuild to resolve from the System GAC", v => {} , true }, // Used in Xamarin.Mac.XM45.targets and must be ignored here. Hidden since it is a total hack. If you can use it, you don't need support
-				{ "disable-lldb-attach=", "Disable automatic lldb attach on crash", v => disable_lldb_attach = ParseBool (v, "disable_lldb_attach")},
+				{ "disable-lldb-attach=", "Disable automatic lldb attach on crash", v => disable_lldb_attach = ParseBool (v, "disable-lldb-attach")},
+				{ "custom-machine-config=", "Custom machine.config file to copy into MonoBundle/mono/4.5/machine.config. Pass \"\" to copy in a valid \"empty\" config file.", v => custom_machine_config_path = v },
```
Comment (rolfbjarne) [inline tools/mmp/driver.cs]: Isn't `custom-` a bit redundant? If you're passing a path here, it's kind of obvious it's a custom file... so maybe name it just `machine-config=` instead?

Comment (chamons) [inline tools/mmp/driver.cs]: Fixed in latest to just --machine-config

File: runtime/launcher.m (hunk 1 above)
Comment (chamons) [inline runtime/launcher.m]: Let me look into this. That is a good point.

**Comment by timrisi** [comment]
Test changes lgtm

File: runtime/launcher.m (hunk 1 above)
Comment (chamons) [inline runtime/launcher.m]: Filed a bug w\ mono here - https://bugzilla.xamarin.com/show_bug.cgi?id=45279

I'll change that setenv param now.

**Comment by chamons** [comment]
Hmm. Somehow i broke msbuild-tests. Will look into it.

**Pull request merged**